### ISSUE TITLE
Make A Set Of Challenges That Guides The Progression More

### ISSUE THREAD

**Issue by opticyclic** [BentoBoxWorld/Challenges#240]
### Description
The current default chest in BSkyBlock contains a lot of items:

    "0": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: LAVA_BUCKET\n",
    "2": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: SUGAR_CANE\n",
    "4": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: RED_MUSHROOM\n",
    "6": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: MELON_SLICE\n",
    "10": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: ICE\n  amount: 2\n",
    "12": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: BONE\n  amount: 2\n",
    "14": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: OAK_SAPLING\n",
    "16": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: PUMPKIN\n",
    "18": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: CACTUS\n",
    "20": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: BROWN_MUSHROOM\n",
    "22": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: COBBLESTONE\n  amount: 5\n",
    "24": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: BEETROOT_SEEDS\n",
    "26": "is:\n  ==: org.bukkit.inventory.ItemStack\n  v: 1631\n  type: MUSHROOM_STEW\n"

Some of these are redundant as they can be got in normal gameplay
e.g. cobblestone, sapling and bone.

A much more simplistic chest would just be one lava bucket and one piece of ice.
The rest of the items would come from rewards through challenges.

#### Describe the solution you'd like us to implement.
The first food you can get is wheat/bread from seeds by breaking the grass.
This farm then allows you to get 64 seeds for the seed collector.
This should then give you a beetroot seed which allows you to start a beetroot farm.
64 beetroot should give you a melon which then allows you to start a melon farm.
A successful melon farm could then give you a pumpkin to start a pumpkin farm.
A pumpkin farm could give you sand.
A carrot farm (from a zombie drop) could give you sugar cane.
A potato farm (from a zombie drop) could give you a cactus.
Then you could make a cactus farm.

You should be able to find both mushroom types in the nether.
However, that is a lot easier to do with a normal nether than a skyblock nether so maybe the cactus farm could give you the mushrooms.

### Additional context
I don't want to make it ultra hard core like ilmango did with no water or lava where he relied on a datapack to get certain items from a wandering trader https://www.youtube.com/watch?v=mMaAOCfrdcM
However, I also don't want to make it too easy right off the bat either.
Certain things should be impossible to do without doing the challenges but those challenges shouldn't be super hard to do.

**Comment by BONNe** [comment]
Well, challenges add-on allows to create things like that. 
But I personally do not want to bother with creating  set of challenges like that. 
Not because I do not like the idea, but because you need to think not only about food items but also other blocks, like andesite and gravel.
And it would still mean that you need to use specific blueprint bundle, as default ones contains all items.
However if you create something like that, I cold definitely add it to web library.

**Comment by FesterHead** [comment]
This seems more of a server customization request than a bug or feature as the capabilities to do this are already present in the BentoBox World.
1. A set of island blueprints that define the starting environment
2. A set of Challenges, which are an optional addon, to import that provide the linear grind

**Comment by opticyclic** [comment]
I agree that the concept of a harder mode does make it harder to implement with the current modularisation because you have to use a specific blueprint bundle with with a specific challenge set and there is currently no way to wrap those together in a single step.

However, I still feel as though the two features could be added to the base BentoBox World since there isn't an overwhelming choice bundled right now (3 blueprints and 2 challenge sets).

The other blocks like andesite and granite would come from the Competent or Advanced level because you can craft them using quartz.
FYI: This craft is something I only found out after playing (A)Skyblock.

Part of the idea behind this is to make you discover crafts and farms that you might not normally be aware of in normal survival and provide the non-craftable items from the challenges.

It isn't clear where the files for the web library are kept and where a pull request should be made.

**Comment by Poslovitch** [comment]
> It isn't clear where the files for the web library are kept and where a pull request should be made.

https://github.com/BentoBoxWorld/weblink

**Comment by opticyclic** [comment]
Can you transfer this issue to that repo then and I'll make a pull request.
